### ISSUE TITLE
[BUG] TMC2209 QUICK_HOME not working with sensor-less homing

### ISSUE THREAD

**Issue by tmartin007** [MarlinFirmware/Marlin#16568]
**Description** 

This is very similar to #10075, which had been closed almost a year ago.

This concerns Marlin-bugfix-2.0.x running on a SKR 1.1V Pro with 2209 drivers, controlling a CNC 6040 - not for milling, but for building a special pick and place robot.

Sensor-less homing is basically working, however:

1) ISSUE
**Steps to Reproduce**
QUICK_HOME is enabled in Configuration_adv.h (attached)
     
When issuing a G28XY ...

**Actual behavior** 
the X- and Y-axis movements happen in sequence 

**Expected behavior** 
rather than simultaneously

2) ISSUE
**Steps to Reproduce** 
X_BED_SIZE and Y_BED_SIZE (thus X_MAX_POS and Y_MAX_POS) are correctly defined.

When issuing a e.g. G0X1000 ...

**Expected behavior**
    Either an error should be reported or the movement should stop at X_MAX_POS, since "1000"  
    exceeds X_MAX_POS (same issue with the Y-axis/Z-axis), alternatively:
    the stallguard should kick in, in the same fashion as it is causing the homing to work correctly.
 
**Actual behavior**
    Instead, all steps are executed regardless, the carriage neither stops  
    at e.g. X_MAX_POS, nor is X_STALL_SENSITIVITIES respected. The carriage bounces right into 
    the physical limit of the machine (and keeps driving the motor until marlin believes that the 
    destination position has been reached).
    The stallguard does not kick in in this situation, which might actually be a third issue.

**Comment by boelle** [comment]
Configurations, please
Please ZIP up your `Configuration.h` and `Configuration_adv.h` files (as
requested in the Issue template) and drop them into your next reply.
We'll check them over and see if anything is amiss.

**Comment by tmartin007** [comment]
[configuration.zip](https://github.com/MarlinFirmware/Marlin/files/4062401/configuration.zip)

Thanks for your prompt comment!

Hopefully, the uploading works this time ...

**Comment by tmartin007** [comment]
ISSUE 1) has suddenly disappeared - I have not changed a thing, though ... - I'll report, should it re-appear. 

ISSUE 2) remains

**Comment by tmartin007** [comment]
Hello,

ISSUE 1) is back
but I found a reproducible pattern:
The issue seems to be related to the X_ and Y_STALL_SENSITIVITY.

With values up to 137, QUICK_HOME works as expected.
With values between 137 and 156, both axis move simultaneously for a while (the lower the value, the longer) until the y-axis suddenly stops, only to resume once the x-axis is done homing.
With values beyond 156, QUICK_HOME does not work, instead both axis move in sequence.

I would prefer a higher sensitivity, since it seems that lower sensitivities yield less accurate homing positions.

ISSUE 2) still remains

**Comment by boelle** [comment]
how often do you update? bugfix is updated almost daily

**Comment by tmartin007** [comment]
Thanks for that hint. - So I have installed yesterdays version of Marlin-bugfix-2.0.x.

Regarding the described issues the behaviour is the same.

There is a **3rd issue** (I already had discovered before updating yesterday):

Oftentimes (about 6 out of 10 times), it takes 2, (sometimes 3) consecutive attempt, before homing works correctly. - On the first and second attempt, Marlin seems to think, the X-, Y- or Z-axis are already homed and just performs the HOMING_BACKOFF_MM on that axis. 

Should I open a separate issue for this, since it does not depend on the QUICK_HOME option ?

**Comment by boelle** [comment]
yep, that would be the best i think

**Comment by boelle** [comment]
btw... even on prusa printers with sensorless homing there is a deviation of about 0.3mm so its not a feature i would recommend

use endstops, they have been proven to work for ages, sensorless is just a gimmick

**Comment by tmartin007** [comment]
Sorry for asking: since this issue is closed, does this mean it is solved - as in "bug fixed" (and I have missed it) ?

**Comment by thinkyhead** [comment]
It is presumed solved. Please confirm.

**Comment by tmartin007** [comment]
I do confirm - this appears to be solved now. - Thanks!
However, #17128 occurred - and might be related ?
